test/boost/error_injection_test.cc

```diff
@@ -40,6 +41,14 @@ SEASTAR_TEST_CASE(test_inject_noop) {
 
     auto f = errinj.inject("noop2", sleep_msec);
     BOOST_REQUIRE(f.available() && !f.failed());
+
+    errinj.enable("noop3");
+    f = errinj.inject_with_handler("noop3", [] (auto& handler) {
+        return handler.wait_for_message(db::timeout_clock::time_point::max());
```

Comment:
Please don't use `max()`, we don't want the test to hang indefinitely if there's a bug, it should throw instead after some time. I suggest using something like `now() + 5s`

(this applies to all cases where you use `max()`)

Reply:
Also, we should ensure that all injections will finish before the test ends. Here the injection would just stay forever. This is problematic - since the error injection object is global, it will be available in other tests and affect them!

For that reason, we should always, at the end of each test case, deliver as many messages as are needed and then wait for the handler to finish.